cmd/openshift-install/internal_integration_test.go

```diff
@@ -103,11 +104,68 @@ func runIntegrationTest(t *testing.T, testFolder string) {
 				}
 			}
 
-			// Let's get the current release version, so that
-			// it could be used within the tests
-			pullspec, err := releaseimage.Default()
-			if err != nil {
-				return err
+			var pullspec string
+			// If set, let's use $OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE to replace test data
+			// and also pass it to the testscript environment. It will be usually set in a CI job
+			// to reference the ephemeral payload release.
+			if releaseImageOverride, ok := os.LookupEnv("OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"); ok && releaseImageOverride != "" {
+				pullspec = releaseImageOverride
+				e.Vars = append(e.Vars, fmt.Sprintf("OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=%s", pullspec))
+				if authFilePath, ok := os.LookupEnv("AUTH_FILE"); ok && authFilePath != "" {
+					// Read the new pull secret from auth.json
+					authFile, err := os.ReadFile(authFilePath)
+					if err != nil {
+						return err
+					}
+					var auths map[string]interface{}
+					if err := json.Unmarshal(authFile, &auths); err != nil {
+						t.Logf("Error unmarshalling auth.json: %v\n", err)
+						return err
+					}
+					newPullSecret, err := json.Marshal(auths)
+					if err != nil {
+						return err
+					}
+
+					// Read the existing install-config.yaml
+					workDir := e.Getenv("WORK")
+					installConfigPathAbs := filepath.Join(workDir, "install-config.yaml")
+					installConfigFile, err := os.ReadFile(installConfigPathAbs)
+					if err != nil {
+						if os.IsNotExist(err) {
+							// skip the error when test does not required reading the install-config.yaml
+							return nil
```

Comment:
I think this is the issue - when we return early, the env var doesn't get set on line 170.
So this should change to

```
    if !os.IsNotExist(err) {
        return err
    }
}  else {
```

(or move all this auth stuff to its own function) and then `replaceReleaseImage()` should not be needed any more.

(This took me approximately 3 hours to find :sweat_smile:)

Reply:
We need to ignore the error above when test does not require reading the install-config.yaml else we get an error such as 
```
oot@hardprov-fx2-21 installer]# ./hack/go-integration-test.sh 
?       github.com/openshift/installer/cmd/node-joiner  [no test files]
--- FAIL: TestAgentIntegration (821.43s)
    --- FAIL: TestAgentIntegration/agent_agentconfigtemplate (0.00s)
        --- FAIL: TestAgentIntegration/agent_agentconfigtemplate/agent-config-template (0.00s)
            testscript.go:558: FAIL: testdata/agent/agentconfigtemplate/agent-config-template.txt:0: open $WORK/install-config.yaml: no such file or directory

```